### ISSUE TITLE
Button text touches outline on focus

### ISSUE THREAD

**Issue by anurag2787** [OWASP/Nest#3150]
**Describe the bug**
The Show more button in the project doesn't have any internal padding causing the text to touch the focus outline Also, the focus outline appears on click or touch interaction which looks visually odd Ideally the outline should only be visible during keyboard navigation.

**To Reproduce**

1. Go to https://nest.owasp.org/organizations/juice-shop
2. Scroll to the Top Contributors section
3. Click the Show more button using a mouse or touch
4. Notice that the focus outline appears and the text touches the outline

**Expected behavior**
- The Show more button should have sufficient padding
- The focus outline should only appear during keyboard navigation
- Mouse click or touch interaction should not trigger the focus outline

**Are you going to work on fixing this?**

- [x] Yes
- [ ] No

**Screenshots**
<img width="279" height="178" alt="Image" src="https://github.com/user-attachments/assets/15026393-5b5e-427d-8e69-d29e5ae8f016" />

**Comment by Sanskriti10247** [comment]
Hey @anurag2787 can i work upon this for now ! Will keep you updated upon the work!? 

@arkid15r let me know if i can take this up as an issue!?
Thanks
